app/controllers/vacation_requests_controller.rb

```diff
@@ -18,9 +17,48 @@ def create
     end
   end
 
+  def show
+    render json: @vacation_request
+  end
+
+  def update
+    if @vacation_request.update(vacation_request_params)
+      head  status: :no_content
+    else
+      head  status: :not_found
+    end
+  end
+
+  def requested
+    requests = VacationRequest.where(user_id: params[:user_id]).requested
+    render json: requests
+  end
+
 private
 
-  def vacation_requests_params
-    params.require(:vacation_request).permit(:kind, :start, :duration)
+  def set_vacation_request
+    @vacation_request = VacationRequest.find params[:id]
+  end
+
+  def vacation_request_params
+    prepare_date_params
+    params.require(:vacation_request)
+      .permit(:kind, :start, :end, :duration, :status)
+  end
+
+  def vacation_params(val)
+    params[:vacation_request][val]
+  end
+
+  def prepare_date_params
+    date_from_ms(:start)
+    date_from_ms(:end)
+  end
+
+  # Convert JS Date/Time in miliseconds to Ruby Date format
+  def date_from_ms(val)
```

Comment:
According to method name it just should to return parsed time, not modify params obj